testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/mail/basic/MailTestCase.java

```diff
@@ -0,0 +1,128 @@
+                "beans.xml");
+    }
+
+    @ArquillianResource
+    private URL url;
+
+    @BeforeClass
+    public static void testRequiresDocker() {
+        AssumeTestGroupUtil.assumeDockerAvailable();
+    }
+
+    static class MailSetup implements ServerSetupTask {
+
+        @Override
+        public void setup(ManagementClient managementClient, String containerId) throws Exception {
+            WaitingConsumer consumer = new WaitingConsumer();
+            apacheJames.followOutput(consumer, OutputFrame.OutputType.STDOUT);
+            consumer.waitUntil(frame -> frame.getUtf8String().contains("AddUser command executed sucessfully"), 30,
+                    TimeUnit.SECONDS, 3);
+
+            String mailHost = apacheJames.getHost();
+            Integer smtpPort = apacheJames.getMappedPort(25);
+            Integer pop3Port = apacheJames.getMappedPort(110);
+
+            ModelNode op = Util.createAddOperation(PathAddress.parseCLIStyleAddress("/subsystem=mail/mail-session=mail-test-basic"));
+            op.get("jndi-name").set("java:jboss/mail/mail-test-basic");
+            op.get("debug").set("true");
+
+            ManagementOperations.executeOperation(managementClient.getControllerClient(), op);
+
+            // SMTP socket bindings
+            op = Util.createAddOperation(PathAddress.parseCLIStyleAddress("/socket-binding-group=standard-sockets/remote-destination-outbound-socket-binding=mail-test-basic-smtp-binding"));
+            op.get("host").set(mailHost);
+            op.get("port").set(smtpPort);
+
+            ManagementOperations.executeOperation(managementClient.getControllerClient(), op);
+
+            // POP3 socket bindings
+            op = Util.createAddOperation(PathAddress.parseCLIStyleAddress("/socket-binding-group=standard-sockets/remote-destination-outbound-socket-binding=mail-test-basic-pop3-binding"));
+            op.get("host").set(mailHost);
+            op.get("port").set(pop3Port);
+
+            ManagementOperations.executeOperation(managementClient.getControllerClient(), op);
+
+            // SMTP Mail server
+            op = Util.createAddOperation(PathAddress.parseCLIStyleAddress("/subsystem=mail/mail-session=mail-test-basic/server=smtp"));
+            op.get("outbound-socket-binding-ref").set("mail-test-basic-smtp-binding");
+            op.get("username").set("user01@james.local");
+            op.get("password").set("1234");
+
+            ManagementOperations.executeOperation(managementClient.getControllerClient(), op);
+
+            // POP3 Mail server
+            op = Util.createAddOperation(PathAddress.parseCLIStyleAddress("/subsystem=mail/mail-session=mail-test-basic/server=pop3"));
+            op.get("outbound-socket-binding-ref").set("mail-test-basic-pop3-binding");
+
+            ManagementOperations.executeOperation(managementClient.getControllerClient(), op);
+
+            // Reload to apply the changes
+            ServerReload.reloadIfRequired(managementClient);
+        }
```

Comment:
Just use `org.jboss.as.test.shared.ManagementServerSetupTask` instead.

Reply:
Ok, got it, thanks @rhusar . The operations don't need to go in a batch. Out of curiosity, what would be the advantage vs the current one?

Reply:
Well, I guess, they are applied faster, any other?

Reply:
ahh, ok, I see it, if one of them fails, there won't be inconsistencies with partially applied operations, that's good

Reply:
Sorry forgot to reply, but yes apart from getting a consistent server config, it's also simple to use and less verbose = less LOC to manage :)